ffi/src/lib.rs

```diff
@@ -1755,3 +1776,275 @@ pub(crate) unsafe fn get_pipeline_options(
         PipelineRetryStrategy::new(info.retry_server_error, info.retry_connection_error),
     )
 }
+
+/// Creates an OpenTelemetry span with the given name and returns a pointer to the span as u64.
+///
+/// # Safety
+/// * `request_type` must be a valid request type.
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn create_otel_span(request_type: RequestType) -> u64 {
+    let cmd = request_type
+        .get_command()
+        .expect("Couldn't fetch command type");
+    let cmd_bytes = match cmd.command() {
+        Some(bytes) => bytes,
+        None => return 0, // Return 0 if no command bytes available
+    };
+    let command_name = match std::str::from_utf8(cmd_bytes.as_slice()) {
+        Ok(name) => name,
+        Err(_) => return 0, // Return 0 if command bytes are not valid UTF-8
+    };
+
+    let span = GlideOpenTelemetry::new_span(command_name);
+    let arc = Arc::new(span);
+    let ptr = Arc::into_raw(arc);
+    ptr as u64
+}
+
+/// Creates an OpenTelemetry span with a fixed name "batch" and returns a pointer to the span as u64.
+///
+/// # Safety
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn create_batch_otel_span() -> u64 {
+    let command_name = "Batch";
+
+    let span = GlideOpenTelemetry::new_span(command_name);
+    let arc = Arc::new(span);
+    let ptr = Arc::into_raw(arc);
+    ptr as u64
+}
+
+/// Drops an OpenTelemetry span given its pointer as u64.
+///
+/// # Safety
+/// * `span_ptr` must be a valid pointer to a span created by [`create_otel_span`].
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn drop_otel_span(span_ptr: u64) {
+    if span_ptr == 0 {
+        return;
+    }
+    unsafe {
+        Arc::from_raw(span_ptr as *const GlideSpan);
+    }
+}
+
+/// Configuration for OpenTelemetry integration in the Node.js client.
+///
+/// This struct allows you to configure how telemetry data (traces and metrics) is exported to an OpenTelemetry collector.
+/// - `traces`: Optional configuration for exporting trace data. If `None`, trace data will not be exported.
+/// - `metrics`: Optional configuration for exporting metrics data. If `None`, metrics data will not be exported.
+/// - `flush_interval_ms`: Optional interval in milliseconds between consecutive exports of telemetry data. If `None`, a default value will be used.
+///
+/// At least one of traces or metrics must be provided.
+#[repr(C)]
+#[derive(Clone)]
```

Comment:
```suggestion
#[derive(Clone, Debug)]
```
This may help you debugging things if needed. Applicable to all structs